### ISSUE TITLE
Feature request: Ask for credentials at Runtime and allow to save in OS' credentials store

### ISSUE THREAD

**Issue by slawomir-brzezinski-at-interxion** [Aqovia/OctopusPuppet#6]
This is so that on infrastructures where OctopusPuppet deployment is centrally managed, the deployment can be a simple delete old/copy new, but the user doesn't lose his credentials (and is not tempted to keep them somewhere unencrypted, or worse, bake them into deployment).

**Comment by taliesins** [comment]
We probably don't want to be storing API credentials as Octopus variables and using them to set configuration for each developers machine.

I think I am going to add Powershell Deployments so that during install time you can make use of environment variables to override Octopus supplied variables when the config file is created.

We could check if there is an environment variable and use this as well. 

We could also store credentials in windows credential manager.

So perhaps we first check windows credential manager and if there is nothing there, then we try environment variable and if there is nothing there, then we fall back to config file?

**Comment by zlamma** [comment]
Yes checking Windows Credential Manager first would definitely be the best preference, since it nicely keeps the credentials unreadable for other users of the machine or anyone gaining access to it outside of user's session.
Environmental variable as a second option seems okay, but I seem to also see a lot of value in addding a user-local config overrides, which at least nudges the user to store the password in a file inaccessible to other users and also because some other users own settings that don't get reset on deployment will soon be needed.
